vendor/k8s.io/kubernetes/staging/src/k8s.io/apiserver/pkg/server/dynamiccertificates/named_certificates.go

```diff
@@ -52,7 +52,7 @@ func (c *DynamicServingCertificateController) BuildNamedCertificates(sniCerts []
 
 		klog.V(2).Infof("loaded SNI cert [%d/%q]: %s", i, c.sniCerts[i].Name(), GetHumanCertDetail(x509Cert))
 		if c.eventRecorder != nil {
-			c.eventRecorder.Eventf(nil, nil, v1.EventTypeWarning, "TLSConfigChanged", "SNICertificateReload", "loaded SNI cert [%d/%q]: %s with explicit names %v", i, c.sniCerts[i].Name(), GetHumanCertDetail(x509Cert), names)
+			c.eventRecorder.Eventf(&corev1.ObjectReference{Name: c.sniCerts[i].Name()}, nil, corev1.EventTypeWarning, "TLSConfigChanged", "SNICertificateReload", "loaded SNI cert [%d/%q]: %s with explicit names %v", i, c.sniCerts[i].Name(), GetHumanCertDetail(x509Cert), names)
```

Comment:
won't this assign the event to the secret and not the operand? Little bit odd.

Reply:
it will reference `clientCA` which is the source of the change so we can trace it back.

Reply:
I think these might be coming from a file

Reply:
so we will know which object in a file has changed.

Reply:
But the "action" comes from the API server, i.e. the actor is the apiserver, the object is the secret. Will the event not look backwards?